### PR TITLE
Hw26

### PR DESCRIPTION
**Pull request by andriylagun**

### CODE REVIEW

**Review by andromash — APPROVED** [review]
GenericDAO and GenericService should have been made in the previous homework, but still a good job!

File: src/main/java/com/internet/shop/controller/RegistrationController.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+import java.io.IOException;
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/register")
+public class RegistrationController extends HttpServlet {
+    private static final Injector INJECTOR = Injector.getInstance("com.internet.shop");
```
Comment (RomanDubovskyi) [inline src/main/java/com/internet/shop/controller/RegistrationController.java]:
```suggestion
    private static final Injector injector = Injector.getInstance("com.internet.shop");
```

File: src/main/java/com/internet/shop/controller/cart/AddProductToCartController.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/shopping-cart/products/add")
+public class AddProductToCartController extends HttpServlet {
+    private static final Long USER_ID = 1L;
+    private static final Injector INJECTOR = Injector.getInstance("com.internet.shop");
```
Comment (RomanDubovskyi) [inline src/main/java/com/internet/shop/controller/cart/AddProductToCartController.java]: same with injector everywhere

File: src/main/webapp/WEB-INF/views/addProduct.jsp [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+            background-size: cover;
+        }
+    </style>
+</head>
+<body>
+<%@include file="header.html"%>
+<div class="container" align="center">
+    <h2>Please enter product name and price:</h2>
+    <p></p>
+    <form method="post" action="/products/add">
```
Comment (RomanDubovskyi) [inline src/main/webapp/WEB-INF/views/addProduct.jsp]:
```suggestion
    <form method="post" action="${pageContext.request.contextPath}/products/add">
```
it's better to use contextPath, so code won't break is you change url settings in tomcat 
same goes for other links

File: src/main/java/com/internet/shop/controller/cart/AddProductToCartController.java (hunk 2 above)
Comment (RomanDubovskyi) [reply src/main/java/com/internet/shop/controller/cart/AddProductToCartController.java]: i can see this comment wasn't fixed yet

File: src/main/webapp/WEB-INF/views/allProducts.jsp [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+                            <c:out value="${product.id}"/>
+                        </th>
+                        <td>
+                            <c:out value="${product.name}"/>
+                        </td>
+                        <td>
+                            <c:out value="${product.price}"/>
+                        </td>
+                        <td>
+                            <a class="btn btn-success btn-small" href="/shopping-cart/products/add?id=${product.id}">Add to cart</a>
```
Comment (RomanDubovskyi) [inline src/main/webapp/WEB-INF/views/allProducts.jsp]: you missed here context path

File: src/main/java/com/internet/shop/controller/cart/AddProductToCartController.java (hunk 2 above)
Comment (andriylagun) [reply src/main/java/com/internet/shop/controller/cart/AddProductToCartController.java]: done

File: src/main/webapp/WEB-INF/views/allProducts.jsp (hunk 4 above)
Comment (andriylagun) [reply src/main/webapp/WEB-INF/views/allProducts.jsp]: done

**Review by RomanDubovskyi — APPROVED** [review]

**Pull request merged**